platform/o.n.swing.laf.flatlaf/src/org/netbeans/swing/laf/flatlaf/ui/FlatTabControlIcon.java

```diff
@@ -0,0 +1,272 @@
+
+    private static Map<Entry<Integer,Integer>,Icon> populateInstances() {
+        // The string keys of these maps aren't currently used, but are useful for debugging.
+        Map<String, Integer> buttonIDs = new LinkedHashMap<String, Integer>();
+        // ViewTabDisplayerUI
+        buttonIDs.put("close", TabControlButton.ID_CLOSE_BUTTON); // NOI18N
+        // These don't seem to be in use anymore.
+        //buttonIDs.put("slide_right", TabControlButton.ID_SLIDE_RIGHT_BUTTON); // NOI18N
+        //buttonIDs.put("slide_left", TabControlButton.ID_SLIDE_LEFT_BUTTON); // NOI18N
+        //buttonIDs.put("slide_down", TabControlButton.ID_SLIDE_DOWN_BUTTON); // NOI18N
+        buttonIDs.put("pin", TabControlButton.ID_PIN_BUTTON); // NOI18N
+        buttonIDs.put("restore_group", TabControlButton.ID_RESTORE_GROUP_BUTTON); // NOI18N
+        buttonIDs.put("slide_group", TabControlButton.ID_SLIDE_GROUP_BUTTON); // NOI18N
+        // EditorTabDisplayerUI
+        buttonIDs.put("scroll_left", TabControlButton.ID_SCROLL_LEFT_BUTTON); // NOI18N
+        buttonIDs.put("scroll_right", TabControlButton.ID_SCROLL_RIGHT_BUTTON); // NOI18N
+        buttonIDs.put("drop_down", TabControlButton.ID_DROP_DOWN_BUTTON); // NOI18N
+        buttonIDs.put("maximize", TabControlButton.ID_MAXIMIZE_BUTTON); // NOI18N
+        buttonIDs.put("restore", TabControlButton.ID_RESTORE_BUTTON); // NOI18N
+        Map<String, Integer> buttonStates = new LinkedHashMap<String, Integer>();
+        buttonStates.put("default", TabControlButton.STATE_DEFAULT); // NOI18N
+        buttonStates.put("pressed", TabControlButton.STATE_PRESSED); // NOI18N
+        buttonStates.put("disabled", TabControlButton.STATE_DISABLED); // NOI18N
+        buttonStates.put("rollover", TabControlButton.STATE_ROLLOVER); // NOI18N
+        Map<Entry<Integer,Integer>,Icon> ret = new LinkedHashMap<Entry<Integer,Integer>,Icon>();
+        for (Entry<String,Integer> buttonID : buttonIDs.entrySet()) {
+          for (Entry<String,Integer> buttonState : buttonStates.entrySet()) {
+              populateOne(ret, buttonID.getValue(), buttonState.getValue());
+          }
+        }
+        // Effectively immutable upon assignment to the final static variable.
+        return Collections.unmodifiableMap(ret);
+    }
+
+    private FlatTabControlIcon(int buttonId, int buttonState) {
+        super(UIScale.scale(16), UIScale.scale(16));
+        this.buttonId = buttonId;
+        this.buttonState = buttonState;
+        this.userScaleFactor = UIScale.getUserScaleFactor();
+    }
+
+    /**
+     * @return null if the requested icon is not available in vector format
+     */
+    public static Icon get(int buttonId, int buttonState) {
+        return INSTANCES.get(new SimpleEntry<Integer,Integer>(buttonId, buttonState));
+    }
+
+    @Override
+    protected void paintIcon(Component c, Graphics2D g, int width, int height, double scaling) {
+        // scale on Java 8 and Linux
+        scaling *= userScaleFactor;
```

Comment:
Have you tested setting a userScaleFactor != 1? Shouldn't the width/height parameters be changed as well in this case?

Reply:
Yes, width and height are scaled in the constructor with `UIScale.scale()`